src/components/ProductCont/ProductCard.jsx

```diff
@@ -0,0 +1,75 @@
+import axios from "axios";
+import {  useEffect } from "react";
+import { useProductContext } from "../context/ProductContext";
+import '../ProductCont/Product_page.css'
+
+
+
+
+export const ProductCard = () => {
+ 
+  const { products, dispatch } = useProductContext();
+
+ 
+  useEffect(() => {
+    
+    
+    (async function () {
+        const { data } = await axios.get("./api/products");
+        dispatch({ type: "GET_PRODUCT", payload: data.products });
+      })();
+  }, );
+ 
+  return (
+  
+        <>
+        <main>
+          {products.map((products) => {
```

Comment:
suggestion - you may destructure `products` in the next line and use those properties. this will reduce you effort in writing `product.` before every property.